src/main/groovy/com/twcable/gradle/sling/SlingServerConfiguration.groovy

```diff
@@ -127,57 +122,65 @@ class SlingServerConfiguration {
     }
 
-    /**
-     * Returns the base URL to control a bundle.
-     */
-    @Nonnull
-    URI getBundleControlBaseUri() {
-        URI base = getBaseUri()
-        new URI(base.scheme, base.userInfo, base.host, base.port, '/system/console/bundles', null, null)
-    }
-
-    /**
-     * Returns the URL to use to do actions on a bundle.
-     */
-    @Nonnull
-    URI getBundleControlUriJson() {
-        URI base = getBundleControlBaseUri()
-        new URI(base.scheme, base.userInfo, base.host, base.port, "${base.path}.json", null, null)
-    }
-
-    /**
-     * Returns the URL to use to do actions on a CQ package.
-     */
-    @Nonnull
-    URI getPackageControlUri() {
-        URI base = getBaseUri()
-        new URI(base.scheme, base.userInfo, base.host, base.port, '/crx/packmgr/service/.json', null, null)
-    }
+//    /**
+//     * Returns the URL to use to do actions on a CQ package.
+//     */
+//    @Nonnull
+//    URI getPackageControlUri() {
+//        URI base = getBaseUri()
+//        new URI(base.scheme, base.userInfo, base.host, base.port, '/crx/packmgr/service/.json', null, null)
+//    }
+//
+//    /**
+//     * Returns the URL to list all the CQ packages.
+//     */
+//    @Nonnull
+//    URI getPackageListUri() {
+//        URI base = getBaseUri()
+//        new URI(base.scheme, base.userInfo, base.host, base.port, '/crx/packmgr/list.jsp', null, null)
+//    }
+//
+//    /**
+//     * Returns the URL to download a CQ package.
+//     */
+//    @Nonnull
+//    URI getPackageDownloadUri() {
+//        URI base = getBaseUri()
+//        new URI(base.scheme, base.userInfo, base.host, base.port, '/crx/packmgr/download.jsp', null, null)
+//    }
+
+    /**
+     * Sets the property to the given value, doing type coercion if needed
+     */
+    void setTheProperty(final String propName, Object value) {
+        String realPropName = propName
+
+        // do any needed translations
+        switch (propName) {
+            case 'machinename': realPropName = 'machineName'; break
+            case 'retry.ms': realPropName = 'retryWaitMs'; break
+            case 'max.ms': realPropName = 'maxWaitMs'; break
+        }
 
-    /**
-     * Returns the URL to list all the CQ packages.
-     */
-    @Nonnull
-    URI getPackageListUri() {
-        URI base = getBaseUri()
-        new URI(base.scheme, base.userInfo, base.host, base.port, '/crx/packmgr/list.jsp', null, null)
-    }
+        if (!this.hasProperty(realPropName)) {
+            throw new IllegalArgumentException("\"${realPropName}\" is not a known property on ${this}")
+        }
 
-    /**
-     * Returns the URL to upload to when installing a bundle.
-     */
-    @Nonnull
-    URI getBaseInstallUri() {
-        final URI base = baseUri
-        new URI(base.scheme, base.userInfo, base.host, base.port, "${base.path}${installPath}", null, null).normalize()
+        final def propertyType = this.metaClass.getMetaProperty(realPropName).type
+
+        switch (propertyType) {
```

Comment:
It can't be blank or null. If the property isn't found, an exception is thrown, and of course every property has a type.